src/earn/EarnEnterAmount.tsx

```diff
@@ -263,10 +225,10 @@ function EarnEnterAmount({ route }: Props) {
     clearPreparedTransactions()
 
     if (
-      !tokenAmount ||
+      !processedAmounts.token.bignum ||
       !transactionTokenAmount ||
-      tokenAmount.isLessThanOrEqualTo(0) ||
-      tokenAmount.isGreaterThan(balanceInInputToken)
+      processedAmounts.token.bignum.isLessThanOrEqualTo(0) ||
+      processedAmounts.token.bignum.isGreaterThan(inputToken.balance)
```

Comment:
perhaps we should add the inputToken to the dependency array...but i see that `balanceInInputToken` wasn't part of it but that may have been because it was derived from `transactionToken`?

Reply:
Ah, you're right!